packages/frontend/editor-ui/src/components/RunDataAi/utils.ts

```diff
@@ -595,6 +579,63 @@ export function flattenLogEntries(
 	return ret;
 }
 
+function sortLogEntries<T extends { runData: ITaskData }>(a: T, b: T) {
+	// We rely on execution index only when startTime is different
+	// Because it is reset to 0 when execution is waited, and therefore not necessarily unique
+	if (a.runData.startTime === b.runData.startTime) {
+		return a.runData.executionIndex - b.runData.executionIndex;
+	}
+
+	return a.runData.startTime - b.runData.startTime;
+}
+
+export function mergeStartData(
+	startData: { [nodeName: string]: ITaskStartedData[] },
+	response: IExecutionResponse,
+): IExecutionResponse {
+	if (!response.data) {
+		return response;
+	}
+
+	const nodeNames = uniq(
+		Object.keys(startData).concat(Object.keys(response.data.resultData.runData)),
+	);
```

Comment:
```suggestion
const nodeNames = [...new Set(Object.keys(startData).concat(Object.keys(response.data.resultData.runData))).values()]
```

you can use a Set here, it's faster and there is no need for a third party dependency, but it's up to you